presto-main/src/main/java/io/prestosql/sql/planner/optimizations/AddExchanges.java

```diff
@@ -206,8 +207,7 @@ public PlanWithProperties visitAggregation(AggregationNode node, PreferredProper
             PreferredProperties preferredProperties = preferSingleNode ? PreferredProperties.undistributed() : PreferredProperties.any();
 
             if (!node.getGroupingKeys().isEmpty()) {
-                preferredProperties = PreferredProperties.partitionedWithLocal(partitioningRequirement, grouped(node.getGroupingKeys()))
-                        .mergeWithParent(parentPreferredProperties);
+                preferredProperties = fetchMergedPreferredProperty(PreferredProperties.partitionedWithLocal(partitioningRequirement, grouped(node.getGroupingKeys())), parentPreferredProperties);
```

Comment:
maybe we could static import `partitionedWithLocal`?

Also break this long line into multiple lines